File: software/examples/vbnet/ExampleThreshold.vb
Imports System
Imports Tinkerforge

Module ExampleThreshold
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change to your UID

    ' Callback subroutine for distance reached callback (parameter has unit mm)
    Sub DistanceReachedCB(ByVal sender As BrickletDistanceIR, ByVal distance As Integer)
        Console.WriteLine("Distance: " + (distance/10.0).ToString() + " cm")
    End Sub

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim dir As New BrickletDistanceIR(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Get threshold callbacks with a debounce time of 10 seconds (10000ms)
        dir.SetDebouncePeriod(10000)

        ' Register distance reached callback to subroutine DistanceReachedCB
        AddHandler dir.DistanceReached, AddressOf DistanceReachedCB

        ' Configure threshold for distance "smaller than 30 cm" (unit is mm)
        dir.SetDistanceCallbackThreshold("<"C, 30*10, 0)

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module

File: software/examples/go/example_threshold.go
package main

import (
	"fmt"
	"github.com/tinkerforge/go-api-bindings/distance_ir_bricklet"
	"github.com/tinkerforge/go-api-bindings/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Distance IR Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	dir, _ := distance_ir_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get threshold receivers with a debounce time of 10 seconds (10000ms).
	dir.SetDebouncePeriod(10000)

	dir.RegisterDistanceReachedCallback(func(distance uint16) {
		fmt.Printf("Distance: %f cm\n", float64(distance)/10.0)
	})

	// Configure threshold for distance "smaller than 30 cm".
	dir.SetDistanceCallbackThreshold('<', 30*10, 0)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
